packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
+
+    /**
+     * Define constructor
+     */
+    constructor(props: CarouselProps) {
+        super(props);
+
+        if (this.props.items) {
+            this.state = {
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : this.props.items.length > 0
+                            ? get(this.props.items[0], "id")
+                            : "",
+            };
+
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(): void {
+        if (this.slides !== this.props.items) {
+            this.updateSlides();
```

Comment:
Curious what happens to local state of the active ID in the case that a new item(s) is pushed in?

Reply:
State of active ID does not change

Reply:
@nalogiudice what happens if I pass a different active Id? The internal state is one thing and my prop reference is different. I think that may be a gap here.